File: delivery/php-sdk/models/receiver/entity/delivery/send_entity.php
<?php
/**
 * @Author Hanqiang
 * @Date 2025/12/9
 * @PackageName: models/receiver/entity/delivery
 * @ClassName: send_entity
 * @Description: 麦芽田配送开放平台SDK - 下单接口实体
 * 提供配送下单接口的请求和响应实体定义
 * @Version 1.0
 */

namespace Maiyatian\Delivery\SDK\models\receiver\entity\delivery;

/**
 * SendRequest 配送下单请求实体
 */
class SendRequest
{
    /**
     * 麦芽田方门店ID
     * @var string
     */
    public $shop_id;

    /**
     * 是否预约单
     * @var bool
     */
    public $is_pre_order;

    /**
     * 是否接驳单
     * @var bool
     */
    public $is_transship;

    /**
     * 期望送达时间（单位：unix 时间戳，精确到秒）
     * @var int
     */
    public $delay_delivery_time;

    /**
     * 期望送达时间（单位：unix 时间）
     * @var int
     */
    public $expect_finish_time;

    /**
     * 小费
     * @var int
     */
    public $tips;

    /**
     * 订单号
     * @var string
     */
    public $source_order_no;

    /**
     * 用户备注
     * @var string
     */
    public $remark;

    /**
     * 发票类型 1-公司 ；2-个人
     * @var string
     */
    public $exp_type;

    /**
     * 快递预约上门揽件开始时间
     * @var int
     */
    public $pickup_start_time;

    /**
     * 快递预约上门揽件结束时间
     * @var int
     */
    public $pickup_end_time;

    /**
     * 门店ID
     * @var string
     */
    public $custid;

    /**
     * 运费付款方式，0-寄付现结；1-寄付月结；2-收方付；3-第三方付
     * @var int
     */
    public $pay_mode;

    /**
     * 发货人信息
     * @var SendSenderInfo
     */
    public $sender;

    /**
     * 收货人信息
     * @var SendReceiverInfo
     */
    public $receiver;

    /**
     * 订单信息
     * @var SendOrderInfo
     */
    public $order_info;

    /**
     * 订单业务标识
     * @var array<string>
     */
    public $tags;

    /**
     * SendRequest constructor.
     *
     * @param string $shopId 麦芽田方门店ID
     * @param string $sourceOrderNo 订单号
     */
    public function __construct(string $shopId, string $sourceOrderNo)
    {
        $this->shop_id = $shopId;
        $this->source_order_no = $sourceOrderNo;
        $this->is_pre_order = false;
        $this->is_transship = false;
        $this->sender = new SendSenderInfo();
        $this->receiver = new SendReceiverInfo();
        $this->order_info = new SendOrderInfo();
        $this->tags = [];
    }
}

/**
 * SendSenderInfo 发货人信息实体（下单接口专用）
 */
class SendSenderInfo
{
    /**
     * 门店名称
     * @var string
     */
    public $name;

    /**
     * 获取验证码手机号
     * @var string
     */
    public $phone;

    /**
     * 客户详细地址
     * @var string
     */
    public $address;

    /**
     * 发货门店地址详情
     * @var string
     */
    public $address_detail;

    /**
     * 经度
     * @var string
     */
    public $longitude;

    /**
     * 纬度
     * @var string
     */
    public $latitude;

    /**
     * 行政区编码（省）
     * @var string
     */
    public $province_code;

    /**
     * 行政区编码（城市）
     * @var string
     */
    public $city_code;

    /**
     * 行政区编码（县区）
     * @var string
     */
    public $district_code;

    /**
     * SendSenderInfo constructor.
     */
    public function __construct()
    {
    }
}

/**
 * SendReceiverInfo 收货人信息实体（下单接口专用）
 */
class SendReceiverInfo
{
    /**
     * 收货人
     * @var string
     */
    public $name;

    /**
     * 收货人电话
     * @var string
     */
    public $phone;

    /**
     * 收货人地址
     * @var string
     */
    public $address;

    /**
     * 收货人地址详情
     * @var string
     */
    public $address_detail;

    /**
     * 经度
     * @var string
     */
    public $longitude;

    /**
     * 纬度
     * @var string
     */
    public $latitude;

    /**
     * 收货人行政区编码（省）
     * @var string
     */
    public $province_code;

    /**
     * 收货人行政区编码（城市）
     * @var string
     */
    public $city_code;

    /**
     * 收货人行政区编码（县区）
     * @var string
     */
    public $district_code;

    /**
     * SendReceiverInfo constructor.
     */
    public function __construct()
    {
    }
}

/**
 * SendOrderInfo 订单信息实体（下单接口专用）
 */
class SendOrderInfo
{
    /**
     * 订单流水号
     * @var int
     */
    public $sn;

    /**
     * 完整订单流水号
     * @var string
     */
    public $full_sn;

    /**
     * 订单来源标识
     * @var string
     */
    public $source;

    /**
     * 订单渠道tag
     * @var string
     */
    public $channel_tag;

    /**
     * 门店名称
     * @var string
     */
    public $channel_name;

    /**
     * 渠道订单号
     * @var string
     */
    public $source_no;

    /**
     * 店铺分类
     * @var string
     */
    public $category;

    /**
     * 商品重量（单位：千克）
     * @var int
     */
    public $weight;

    /**
     * 合计费用（折扣后）（单位：分）
     * @var int
     */
    public $total_fee;

    /**
     * 实付金额(分)
     * @var int
     */
    public $paid_fee;

    /**
     * 商品列表
     * @var array<SendGoodsInfo>
     */
    public $goods_list;

    /**
     * SendOrderInfo constructor.
     */
    public function __construct()
    {
        $this->goods_list = [];
    }

    /**
     * 添加商品信息
     *
     * @param SendGoodsInfo $goodsInfo 商品信息
     */
    public function addGoodsInfo(SendGoodsInfo $goodsInfo): void
    {
        $this->goods_list[] = $goodsInfo;
    }

    /**
     * 批量添加商品信息
     *
     * @param array<SendGoodsInfo> $goodsInfos 商品信息列表
     */
    public function addGoodsInfos(array $goodsInfos): void
    {
        foreach ($goodsInfos as $goodsInfo) {
            $this->addGoodsInfo($goodsInfo);
        }
    }
}

/**
 * SendGoodsInfo 商品信息实体（下单接口专用）
 */
class SendGoodsInfo
{
    /**
     * 商品名称
     * @var string
     */
    public $name;

    /**
     * 商品数量
     * @var int
     */
    public $number;

    /**
     * 商品价格
     * @var int
     */
    public $price;

    /**
     * SendGoodsInfo constructor.
     *
     * @param string $name 商品名称
     * @param int $number 商品数量
     * @param int $price 商品价格
     */
    public function __construct(string $name, int $number, int $price)
    {
        $this->name = $name;
        $this->number = $number;
        $this->price = $price;
    }
}

/**
 * SendResponse 配送下单响应实体
 */
class SendResponse
{
    /**
     * 状态码
     * @var int
     */
    public $code;

    /**
     * 响应消息
     * @var string
     */
    public $message;

    /**
     * 业务数据
     * @var SendResponseData
     */
    public $data;
}

/**
 * SendResponseData 配送下单响应数据实体
 */
class SendResponseData
{
    /**
     * 运费（分）
     * @var int
     */
    public $pay_amount;

    /**
     * 优惠费用(分)
     * @var int
     */
    public $coupon;

    /**
     * 配送距离（米）
     * @var int
     */
    public $distance;

    /**
     * 当前溢价(分)
     * @var int
     */
    public $premium;

    /**
     * 重量（克）
     * @var int
     */
    public $weight;

    /**
     * 超重（克）
     * @var int
     */
    public $overweight;

    /**
     * 加费(分)
     * @var int
     */
    public $tips;

    /**
     * 计费时间
     * @var int
     */
    public $at_time;

    /**
     * 期望送达时间,如遇没有可传0
     * @var int
     */
    public $expect_time;

    /**
     * 三方运单号
     * @var string
     */
    public $order_no;

    /**
     * 麦芽田订单号
     * @var string
     */
    public $source_order_no;

    /**
     * 服务包
     * @var string
     */
    public $service_pkg;

    /**
     * 服务标签
     * @var string
     */
    public $service_tag;
}

File: delivery/go-sdk/models/receiver/entity/delivery/send_entity.go
/**
 * @Author Hanqiang
 * @Date 2025/12/9
 * @PackageName: entity
 * @ClassName: send_entity
 * @Description: 配送下单实体定义
 * @Version 1.0
 */

package entity

// SendReq 配送下单请求参数
// command: send
// 必接: 是
// 说明: 麦芽田平台主动调用三方配送服务创建配送订单
type SendReq struct {
	// ShopId 麦芽田方门店ID
	// 麦芽田平台的门店唯一标识
	ShopId string `json:"shop_id"`

	// IsPreOrder 是否预约单
	// true: 预约单; false: 即时单
	IsPreOrder bool `json:"is_pre_order"`

	// IsTransship 是否接驳单
	// true: 是接驳单; false: 不是接驳单
	IsTransship bool `json:"is_transship,optional"`

	// DelayDeliveryTime 期望送达时间
	// Unix时间戳（秒）
	DelayDeliveryTime int64 `json:"delay_delivery_time"`

	// ExpectStartTime 期望开始时间
	// Unix时间戳（秒），可选
	ExpectStartTime int64 `json:"expect_start_time,optional"`

	// ExpectFinishTime 期望完成时间
	// Unix时间戳（秒）
	ExpectFinishTime int64 `json:"expect_finish_time"`

	// Tips 小费
	// 单位：分
	Tips int64 `json:"tips"`

	// SourceOrderNo 订单号
	// 麦芽田平台的订单号
	SourceOrderNo string `json:"source_order_no,optional"`

	// Remark 用户备注
	// 订单备注信息
	Remark string `json:"remark"`

	// Sender 发货人信息
	// 发货门店的详细信息
	Sender SenderInfo `json:"sender"`

	// Receiver 收货人信息
	// 收货客户的详细信息
	Receiver ReceiverInfo `json:"receiver"`

	// OrderInfo 订单信息
	// 订单商品等详细信息
	OrderInfo OrderInfo `json:"order_info"`

	// ExpType 快递类型
	// 1: 非地图类快递; 2: 地图类快递
	ExpType string `json:"exp_type,optional"`

	// PickupStartTime 快递预约上门揽件开始时间
	// Unix时间戳（秒）
	PickupStartTime int64 `json:"pickup_start_time,optional"`

	// PickupEndTime 快递预约上门揽件结束时间
	// Unix时间戳（秒）
	PickupEndTime int64 `json:"pickup_end_time,optional"`

	// Custid 门店ID
	// 客户门店标识
	Custid string `json:"custid,optional"`

	// PayMode 运费付款方式
	// 0: 寄付现结; 1: 寄付月结; 2: 收方付; 3: 第三方付
	PayMode int `json:"pay_mode,optional"`

	// Tags 订单业务标识
	// []string类型，用于标记订单的特殊属性
	Tags []string `json:"tags,optional"`

	// BillingNo 计价单号
	// 从计费接口返回的计价唯一标识，用于锁定计费
	BillingNo string `json:"billing_no,optional"`
}

// SenderInfo 发货人信息
type SenderInfo struct {
	// Name 发货门店名称
	Name string `json:"name"`

	// Phone 发货门店电话
	Phone string `json:"phone"`

	// Address 发货门店地址
	Address string `json:"address"`

	// AddressDetail 发货门店地址详情
	AddressDetail string `json:"address_detail"`

	// Longitude 经度
	// 国测局02标准（如高德地图坐标系）
	Longitude string `json:"longitude"`

	// Latitude 纬度
	// 国测局02标准（如高德地图坐标系）
	Latitude string `json:"latitude"`

	// ProvinceCode 省级行政区编码
	ProvinceCode string `json:"province_code,optional"`

	// CityCode 市级行政区编码
	CityCode string `json:"city_code,optional"`

	// DistrictCode 区县级行政区编码
	DistrictCode string `json:"district_code,optional"`
}

// ReceiverInfo 收货人信息
type ReceiverInfo struct {
	// Name 收货人姓名
	Name string `json:"name"`

	// Phone 收货人电话
	Phone string `json:"phone"`

	// Address 收货人地址
	Address string `json:"address"`

	// AddressDetail 收货人地址详情
	AddressDetail string `json:"address_detail"`

	// Longitude 经度
	// 国测局02标准（如高德地图坐标系）
	Longitude string `json:"longitude"`

	// Latitude 纬度
	// 国测局02标准（如高德地图坐标系）
	Latitude string `json:"latitude"`

	// ProvinceCode 省级行政区编码
	ProvinceCode string `json:"province_code,optional"`

	// CityCode 市级行政区编码
	CityCode string `json:"city_code,optional"`

	// DistrictCode 区县级行政区编码
	DistrictCode string `json:"district_code,optional"`
}

// OrderInfo 订单信息
type OrderInfo struct {
	// Sn 订单流水号
	// 纯数字编号
	Sn int64 `json:"sn"`

	// FullSn 完整订单流水号
	// 包含日期标识的完整流水号，示例: "101_1230"
	FullSn string `json:"full_sn"`

	// Source 订单来源标识
	// 可选值: meituan, eleme, douyin, kuaishou, wechat, alipay等
	// 详见订单来源枚举
	Source string `json:"source"`

	// ChannelTag 订单渠道tag
	// 渠道标识
	ChannelTag string `json:"channel_tag"`

	// ChannelName 渠道名称
	// 订单店铺名称
	ChannelName string `json:"channel_name"`

	// SourceNo 渠道订单号
	// 第三方渠道的原始订单号
	SourceNo string `json:"source_no"`

	// Category 店铺分类
	// 可选值: xiaochi, dangao, shuiguo, naicha等
	// 详见店铺分类枚举
	Category string `json:"category"`

	// Weight 商品重量
	// 单位：千克
	Weight int `json:"weight"`

	// TotalFee 合计费用（折扣后）
	// 单位：分
	TotalFee int64 `json:"total_fee"`

	// PaidFee 实付金额
	// 单位：分
	PaidFee int64 `json:"paid_fee"`

	// IsFromDoor 是否上门取件
	// 0: 不上门取件; 1: 上门取件
	IsFromDoor int `json:"is_from_door,optional"`

	// IsToDoor 是否送件上门
	// 0: 不送件上门; 1: 送件上门
	IsToDoor int `json:"is_to_door,optional"`

	// GoodsList 商品列表
	// 订单中的所有商品信息
	GoodsList []*GoodsInfo `json:"goods_list"`
}

// GoodsInfo 商品信息
type GoodsInfo struct {
	// Name 商品名称
	Name string `json:"name"`

	// Number 商品数量
	Number int `json:"number"`

	// Price 商品价格
	// 单位：分
	Price int64 `json:"price"`

	// Size 商品规格
	// 如: "大杯"、"中杯"等
	Size string `json:"size"`
}

// SendResp 配送下单响应参数
type SendResp struct {
	// OrderNo 三方运单号
	// 配送服务商的订单号
	OrderNo string `json:"order_no"`

	// SourceOrderNo 麦芽田订单号
	// 麦芽田平台的订单号
	SourceOrderNo string `json:"source_order_no"`

	// PayAmount 运费
	// 单位：分
	PayAmount int64 `json:"pay_amount"`

	// Coupon 优惠费用
	// 单位：分，如遇没有可传0
	Coupon int64 `json:"coupon"`

	// Premium 当前溢价
	// 单位：分，如遇没有可传0
	Premium int64 `json:"premium"`

	// Tips 加费
	// 单位：分，如遇没有可传0
	Tips int64 `json:"tips"`

	// Distance 配送距离
	// 单位：米
	Distance int64 `json:"distance"`

	// Weight 重量
	// 单位：克
	Weight int `json:"weight"`

	// Overweight 超重
	// 单位：克
	Overweight int `json:"overweight"`

	// AtTime 计费时间
	// Unix时间戳（秒）
	AtTime int64 `json:"at_time"`

	// ExpectTime 期望送达时间
	// Unix时间戳（秒），如遇没有可传0
	ExpectTime int64 `json:"expect_time"`

	// ServicePkg 服务包内容
	// 示例: "xxx"
	ServicePkg string `json:"service_pkg,optional"`

	// ServiceTag 服务标签
	// 示例: "专人直送"
	ServiceTag string `json:"service_tag,optional"`
}
